prose_core_client_ffi/src/account_observer.rs

```diff
@@ -0,0 +1,12 @@
+use crate::Roster;
+
+use super::types::message::Message;
+
+#[allow(non_snake_case)]
+pub trait AccountObserver: Send + Sync {
+    fn didConnect(&self);
```

Comment:
Is the use of camel case done on purpose here? Eg. exposing a camel-case FFI API? As Rust coding patterns / community conventions enforce snake-case.

Reply:
Yes, the `AccountObserver` is a trait that needs to be implemented by a consumer of the FFI library. We'd either force snake-case everywhere in the FFI or stick to camel case. What's your preference?

Reply:
If I get this right, the `uniffi` crate would then map those declared Rust names to eg. Swift names for the referred-to function. If that's so, then we should keep it camel in the Rust code yes, unless `uniffi` provides a way to "camelify" snake-case names? That way we can stick to pure-Rust conventions in our Rust code.

Reply:
That's how the corresponding Swift code looks like…

```swift
extension ViewModel: AccountObserver {
  func didConnect() {
    print("Did connect")
  }

  func didDisconnect() {
    print("Did disconnect")
  }

  func didReceiveMessage(message: ProseCoreClientFFI.Message) {
    DispatchQueue.main.async {
      self.messages.append(Message(from: message.from, body: message.body))
    }
  }

  func didReceiveRoster(roster: Roster) {
    print(roster)
  }
}
```

Reply:
I'm not sure that converting the case-style on the fly would even work. I haven't looked to deep in the innards of the UniFFI implementation but I'd imaging that the function calls will still have to access the good old exported C symbols. At least for the Swift output there is [no such option](https://mozilla.github.io/uniffi-rs/swift/configuration.html).

Reply:
Let's stick to camel-case in the Rust code then 🙂